### PR TITLE
V5 Docs - `Store` and `MemoryStore`

### PR DESCRIPTION
**Pull request by colincasey**
Rendered pages:
- https://github.com/salesforce/tough-cookie/blob/v5-docs-stores/api/docs/tough-cookie.store.md
- https://github.com/salesforce/tough-cookie/blob/v5-docs-stores/api/docs/tough-cookie.memorycookiestore.md

---
Main PR:
- #384

### CODE REVIEW

File: api/docs/tough-cookie.store.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+## Store class
+
+Base class for [CookieJar](./tough-cookie.cookiejar.md) stores.
+
+The storage model for each [CookieJar](./tough-cookie.cookiejar.md) instance can be replaced with a custom implementation. The default is [MemoryCookieStore](./tough-cookie.memorycookiestore.md)<!-- -->.
+
+**Signature:**
+
+```typescript
+export declare class Store
```
Comment (wjhsf) [inline api/docs/tough-cookie.store.md:14]: This doesn't seem all that helpful. 😅

File: api/docs/tough-cookie.memorycookiestore.md [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+</th><th>
+
+Description
+
+
+</th></tr></thead>
+<tbody><tr><td>
+
+[(constructor)()](./tough-cookie.memorycookiestore._constructor_.md)
```
Comment (wjhsf) [inline api/docs/tough-cookie.memorycookiestore.md:36]: `(constructor)` is kinda weird - is it a convention from the doc generator?

File: lib/memstore.ts [hunk 3]
```diff
@@ -48,27 +17,66 @@ export type MemoryCookieStoreIndex = {
...
     this.synchronous = true
     this.idx = Object.create(null) as MemoryCookieStoreIndex
   }
 
+  /**
+   * Retrieve a {@link Cookie} with the given `domain`, `path`, and `key` (`name`). The RFC maintains that exactly
+   * one of these cookies should exist in a store. If the store is using versioning, this means that the latest or
+   * newest such cookie should be returned.
+   *
+   * Callback takes an error and the resulting Cookie object. If no cookie is found then null MUST be passed instead (that is, not an error).
```
Comment (wjhsf) [inline lib/memstore.ts:50]:
```suggestion
```
No callback in this signature!

image

File: lib/memstore.ts [hunk 4]
```diff
@@ -48,27 +17,66 @@ export type MemoryCookieStoreIndex = {
   }
 }
 
+/**
+ * An in-memory {@link Store} implementation for {@link CookieJar}. This is the default implementation used by
+ * {@link CookieJar} and supports both async and sync operations. Also supports serialization, getAllCookies, and removeAllCookies.
```
Comment (wjhsf) [inline lib/memstore.ts:22]: Should we warn "this is mostly a development tool, you should use something better in production"?

File: lib/memstore.ts [hunk 5]
```diff
@@ -48,27 +17,66 @@ export type MemoryCookieStoreIndex = {
...
+ * @public
+ */
 export class MemoryCookieStore extends Store {
+  /**
+   * This value is `true` since {@link MemoryCookieStore} implements synchronous functionality.
+   */
   override synchronous: boolean
```
Comment (wjhsf) [inline lib/memstore.ts:29]:
```suggestion
  override synchronous: true
```
Not super important, but does this work? 🤷

File: api/docs/tough-cookie.store.md (hunk 1 above)
Comment (colincasey) [reply api/docs/tough-cookie.store.md:14]: Perhaps, but it is accurate 😄

File: lib/memstore.ts (hunk 4 above)
Comment (colincasey) [reply lib/memstore.ts:22]: Does that even need to be stated? Also, there are some cases where you might have a use for an in-memory store in production. For example, if you are an automation tool that needs to login to a site and store an auth cookie.

File: lib/memstore.ts (hunk 5 above)
Comment (colincasey) [reply lib/memstore.ts:29]: No, it breaks the tests.

**Pull request merged**